src/discussions/comments/CommentsViewContainer.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router';
+import CommentsView from './CommentsView';
+import { selectPostComment, selectPostReplies } from './data/selectors';
+import { fetchComment, fetchPostReplies } from './data/thunks';
+
+function CommentsViewContainer() {
+  const { postId } = useParams();
+  const dispatch = useDispatch();
+  const comment = useSelector(selectPostComment(postId));
+  const replies = useSelector(selectPostReplies(postId));
+  useEffect(() => {
+    // The courseId from the URL is the course we WANT to load.
+    dispatch(fetchComment(postId));
+    dispatch(fetchPostReplies(postId));
+  }, [postId]);
+
+  return (
+    <>
+      {comment && (
+        <CommentsView comment={comment} replies={replies} />
+      )}
+    </>
+  );
```

Comment:
While these render similarly, I think the way this was originally written is the pattern typically followed in our existing MFEs.
Assuming that's the case, I'd be inclined to keep this syntax consistent. It makes it easier for new devs to approach, the more consistent it is.

Also, I have a _rational_ dislike of ternary operators ;)

@davidjoy Is that actually the existing pattern or am i mis-remembering?

Reply:
I think this whole thing can likely be boiled down to:

```
return comment && (
  <CommentsView comment={comment} replies={replies} />
);
```

Also not a fan of ternary operators in JSX, as a personal preference; mostly because I just find it hard to read.  I wasn't a fan of _any_ logic in the middle of JSX, but it's common enough that I've relented a bit.